src/backend/distributed/utils/metadata_cache.c

```diff
@@ -276,9 +278,16 @@ LookupDistTableCacheEntry(Oid relationId)
 			colocationId = INVALID_COLOCATION_ID;
 		}
 
+		replicationModelDatum = heap_getattr(distPartitionTuple,
+											 Anum_pg_dist_partition_repmodel,
+											 tupleDescriptor,
+											 &isNull);
+		Assert(!isNull);
```

Comment:
This assert fails on the upgrades. For colocation, we changed ASSERT to if. We may want to do same here. @aamederen @marcocitus